osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,764 @@
+             << " referenced memory at " << memAddrStr.str() << ".";
+    switch (ex.ExceptionInformation[0]) {
+    case 0:
+      errorMsg << " The memory could not be read.";
+      break;
+    case 1:
+      errorMsg << " The memory could not be written.";
+      break;
+    case 8:
+      errorMsg << " DEP access violation.";
+      break;
+    }
+    r["exception_message"] = errorMsg.str();
+    return Status(0);
+  }
+
+  return Status(1);
+}
+
+Status logRegisters(const MINIDUMP_EXCEPTION_STREAM* stream,
+                    unsigned char* const dumpBase,
+                    Row& r) {
+  if (stream == nullptr)
+    return Status(1);
+
+  auto ex = stream->ExceptionRecord;
+  auto threadContext =
+      reinterpret_cast<CONTEXT*>(dumpBase + stream->ThreadContext.Rva);
+  std::ostringstream registers;
+  // Registers are hard-coded for x64 system b/c lack of C++ reflection on
+  // CONTEXT object
+  registers << "rax:0x" << std::hex << threadContext->Rax;
+  registers << " rbx:0x" << std::hex << threadContext->Rbx;
+  registers << " rcx:0x" << std::hex << threadContext->Rcx;
+  registers << " rdx:0x" << std::hex << threadContext->Rdx;
+  registers << " rdi:0x" << std::hex << threadContext->Rdi;
+  registers << " rsi:0x" << std::hex << threadContext->Rsi;
+  registers << " rbp:0x" << std::hex << threadContext->Rbp;
+  registers << " rsp:0x" << std::hex << threadContext->Rsp;
+  registers << " r8:0x" << std::hex << threadContext->R8;
+  registers << " r9:0x" << std::hex << threadContext->R9;
+  registers << " r10:0x" << std::hex << threadContext->R10;
+  registers << " r11:0x" << std::hex << threadContext->R11;
+  registers << " r12:0x" << std::hex << threadContext->R12;
+  registers << " r13:0x" << std::hex << threadContext->R13;
+  registers << " r14:0x" << std::hex << threadContext->R14;
+  registers << " r15:0x" << std::hex << threadContext->R15;
+  registers << " rip:0x" << std::hex << threadContext->Rip;
+  registers << " segcs:0x" << std::hex << threadContext->SegCs;
+  registers << " segds:0x" << std::hex << threadContext->SegDs;
+  registers << " seges:0x" << std::hex << threadContext->SegEs;
+  registers << " segfs:0x" << std::hex << threadContext->SegFs;
+  registers << " seggs:0x" << std::hex << threadContext->SegGs;
+  registers << " segss:0x" << std::hex << threadContext->SegSs;
+  registers << " eflags:0x" << std::hex << threadContext->EFlags;
+  r["registers"] = registers.str();
+  return Status(0);
+}
+
+Status logPID(const MINIDUMP_MISC_INFO* stream, Row& r) {
+  if ((stream == nullptr) || !(stream->Flags1 & MINIDUMP_MISC1_PROCESS_ID))
+    return Status(1);
+
+  r["pid"] = BIGINT(stream->ProcessId);
+  return Status(0);
+}
+
+Status logProcessCreateTime(const MINIDUMP_MISC_INFO* stream, Row& r) {
+  if ((stream == nullptr) || !(stream->Flags1 & MINIDUMP_MISC1_PROCESS_TIMES))
+    return Status(1);
+
+  time_t procTimestamp = stream->ProcessCreateTime;
+  struct tm gmt;
+  char timeBuff[64];
```

Comment:
considering you're using this below for string formatting, would you mind initializing this memory with something like `memset`?